datafusion/expr/src/udf.rs

```diff
@@ -227,6 +224,19 @@ impl ScalarUDF {
         self.inner.invoke_batch(args, number_rows)
     }
 
+    /// Invoke the function on `args`, returning the appropriate result.
+    ///
+    /// See [`ScalarUDFImpl::invoke_batch`] for more details.
+    pub fn invoke_batch_with_return_type(
+        &self,
+        args: &[ColumnarValue],
+        number_rows: usize,
+        return_type: &DataType,
```

Comment:
How about we add `InvokeUDFArg` struct that includes these to avoid breaking change in the future?

Reply:
Do you think we should roll all the values into this struct or just number_rows and return_type?

Reply:
including args

Reply:
Any thoughts on the new function name, if we roll these in together. Since this subsumes `invoke_batch`, would we then want to deprecate that too?

Reply:
but why do you need `return_type`